src/main/java/seedu/address/model/project/Deadline.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.model.project;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a Project's deadline in TaskHub.
+ * Guarantees: immutable; is always valid
+ */
+public class Deadline {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Project deadline should be a valid date in the dd/MM/yyyy format.\n Example: 17/02/2009";
+
+    /**
+     * Either dd/MM/yyyy or empty string
+     */
+    public static final String VALIDATION_REGEX =
+            "^(([0]?[1-9]|[1|2][0-9]|[3][0|1])[/]([0]?[1-9]|[1][0-2])[/]([0-9]{4}))?$";
+
+    public final String value;
+
```

Comment:
This is just a suggestion but what if we stored a `LocalDate` or `LocalDateTime` object instead of a String for the `value`? In that case, the deadline validity check need only be done once in the constructor and the `LocalDate.of` method need only be run once (instead of every time the `getLocalDate` method is called). This is somewhat similar to how a deadline was stored in a deadline task in our iP.

Reply:
I understand where you're coming from. To solve the addition/removal issue, `ArrayList<LocalDate>` or `Optional<LocalDate>` could have been used (the latter being much more appropriate) but, for now using `String` should suffice.